src/components/Footer.jsx

```diff
@@ -78,14 +78,17 @@ FooterColumn.defaultProps = {
   internal: false,
 };
 
-function Footer(props, ref) {
+function Footer({ footerWidth }) {
   const { t } = useTranslation();
+  const rightMargin = footerWidth === '100%' ? 'auto' : 0;
 
   return (
-    <Box sx={{ bgcolor: 'background.secondary' }} ref={ref}>
-      <Container maxWidth="lg" sx={{ py: 4 }}>
+    <Box sx={{ bgcolor: 'background.secondary' }}>
+      <Container
+        maxWidth="lg"
+        sx={{ py: 4, width: { xs: '100%', md: footerWidth }, mr: rightMargin }}>
```

Comment:
`width: { xs: '100%', md: footerWidth }` is to avoid weird behavior on mobile when closing side panel.

Reply:
Have you considered leaving the footer on top of everything and instead adding a buffer at the bottom of the drawer with the height of the footer?